tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/AspNetCoreTestFixture.cs

```diff
@@ -190,7 +197,20 @@ private async Task EnsureServerStarted(bool sendHealthCheck)
                 {
                     try
                     {
-                        serverReady = await SubmitRequest("/alive-check") == HttpStatusCode.OK;
+                        var dateTime = DateTime.UtcNow;
+                        var respondedOk = await SubmitRequest("/alive-check") == HttpStatusCode.OK;
+                        if (respondedOk)
+                        {
+                            if (waitForHealthcheckSpan)
+                            {
+                                Agent.WaitForSpans(1, minDateTime: dateTime);
+                                serverReady = true;
+                            }
+                            else
+                            {
+                                serverReady = true;
+                            }
+                        }
```

Comment:
I'm not a fan of this code or the extra options it requires 🤔. I think we can remove it and achieve the same thing in a test by simply waiting for a normal span?

Reply:
so the problem I faced was that the `OverheadController` only allows one concurrent request.  And sometimes, the alive-check gets in the way and is the one being selected by the controller and the actual integration tests request weren't getting api security information 😬

Reply:
so here I wanted to make sure the alive-check ping is behind us to actually start a request, to make sure it doesnt interfere and other concurrent requests become rejected for api security.. not sure if I'm clear.. 
I dont see a better solution 🤔

Reply:
Oh I see, it's because even though we've got the _response_ from the server, we haven't "finished" the code path on the server, so we potentially get overlap and hit the controller.

In which case, I think your changes are fine, but perhaps we should just remove the `waitForHealthcheckSpan` option entirely, and _always_ wait for it? I can't see it adding much in the way of overhead to the tests (it's just the time it takes to flush the span really) and it would be better to be consistent IMO?

Reply:
generalized for all tests in a5602d6a758312da7234698a1997fedd25879199